firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionLifecycleNotifier.java

```diff
@@ -20,14 +20,28 @@
 import androidx.annotation.GuardedBy;
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import com.google.android.gms.tasks.SuccessContinuation;
 import com.google.android.gms.tasks.Task;
 import com.google.android.gms.tasks.TaskCompletionSource;
 import com.google.android.gms.tasks.Tasks;
 import java.util.ArrayDeque;
 import java.util.Queue;
+import java.util.concurrent.Executor;
 
 class FirebaseAppDistributionLifecycleNotifier implements Application.ActivityLifecycleCallbacks {
 
+  /** An {@link Executor} that runs tasks on the current thread. */
+  private static final Executor DIRECT_EXECUTOR = Runnable::run;
+
+  /** A functional interface for a function that takes an activity and does something with it. */
+  interface ActivityConsumer<T> {
+    void consume(Activity activity);
+  }
+
+  /** A {@link SuccessContinuation} that takes an activity and returns a new {@link Task}. */
+  interface ActivityContinuation<T> extends SuccessContinuation<Activity, T> {}
```

Comment:
Leveraging the Task API's SuccessContinuation interface here.

Reply:
I can't decide if extending `SuccessContinuation` to bind to a specific type makes it cleaner (because there's one less type to worry about) or makes my head hurt (because I have to remember that there's an `is a` relationship here). I'm ok keeping it like this if y'all think it is better. 🙃

Reply:
Ah, you're right I actually like just keeping it as a SuccessContinuation. We're still enforcing that it accepts an Activity. No need to add another type 👍 

Updated.